calico/felix/test/test_config.py

```diff
@@ -273,4 +283,115 @@ def test_ip_in_ip_enabled_bad(self):
         with self.assertRaisesRegexp(ConfigException,
                                      "Field was not a valid Boolean"
                                      ".*IpInIpEnabled"):
-            config.report_etcd_config({}, cfg_dict)
\ No newline at end of file
+            config.report_etcd_config({}, cfg_dict)
+
+    def test_reporting_ttl_not_int(self):
+        """
+        Test exception is rased if status report ttl has invalid (non-integer) value.
+        """
+        with mock.patch('calico.common.complete_logging'):
+            config = Config("calico/felix/test/data/felix_missing.cfg")
+        cfg_dict = { "InterfacePrefix": "blah",
+                     "ReportingTTLSecs": "NaN"}
+        with self.assertRaisesRegexp(ConfigException,
+                                     "Field was not integer.*"):
```

Comment:
Your exception raising code in config.py claims to raise an exception with the message 'Reporting TTL was not integer'. If your test is hitting a 'Field was not integer' exception then it seems there is already more generic error handling code and you do not need the integer checking for ttl/interval in config.py. Alternatively, your test is not working.